paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ach/USBankAccountFormViewModel.kt

```diff
@@ -0,0 +1,405 @@
+                    ConfirmSetupIntentParams.create(
+                        clientSecret = clientSecret.value,
+                        paymentMethodType = PaymentMethod.Type.USBankAccount
+                    )
+                }
+            }
+            _currentScreenState.update {
+                USBankAccountFormScreenState.ConfirmIntent(intent)
+            }
+        }
+    }
+
+    private fun buildPrimaryButtonText(): String? {
+        return when {
+            args.isPaymentSheet -> {
+                if (args.clientSecret is PaymentIntentClientSecret) {
+                    args.formArgs.amount?.buildPayButtonLabel(application.resources)
+                } else {
+                    application.getString(
+                        R.string.stripe_setup_button_label
+                    )
+                }
+            }
+            else -> application.getString(
+                R.string.us_bank_account_payment_sheet_primary_button_continue
+            )
+        }
+    }
+
+    private fun buildMandateText(): String {
+        return if (saveForFutureUse.value) {
+            application.getString(
+                R.string.us_bank_account_payment_sheet_mandate_save,
+                formattedMerchantName()
+            )
+        } else {
+            application.getString(
+                R.string.us_bank_account_payment_sheet_mandate_continue
+            )
+        }
+    }
+
+    internal class Factory(
+        private val applicationSupplier: () -> Application,
+        private val argsSupplier: () -> Args,
+        owner: SavedStateRegistryOwner,
+        defaultArgs: Bundle? = null
+    ) : AbstractSavedStateViewModelFactory(owner, defaultArgs),
+        Injectable<Factory.FallbackInitializeParam> {
+        internal data class FallbackInitializeParam(
+            val application: Application,
+        )
+
+        @Inject
+        lateinit var subComponentBuilderProvider:
+            Provider<USBankAccountFormViewModelSubcomponent.Builder>
+
+        @Suppress("UNCHECKED_CAST")
+        override fun <T : ViewModel> create(
+            key: String,
+            modelClass: Class<T>,
+            savedStateHandle: SavedStateHandle
+        ): T {
+            val args = argsSupplier()
+
+            injectWithFallback(args.injectorKey, FallbackInitializeParam(applicationSupplier()))
+
+            return subComponentBuilderProvider.get()
+                .configuration(args)
+                .savedStateHandle(savedStateHandle)
+                .build().viewModel as T
+        }
+
+        override fun fallbackInitialize(arg: FallbackInitializeParam) {
+            DaggerUSBankAccountFormComponent
+                .builder()
+                .application(arg.application)
+                .injectorKey(DUMMY_INJECTOR_KEY)
+                .build().inject(this)
+        }
+    }
+
+    data class Args(
+        val formArgs: FormFragmentArguments,
+        val isPaymentSheet: Boolean,
```

Comment:
Can you add comments here? `isPaymentSheet` is not very clear since this is always inside PaymentSheet.
If you mean whether it's the complete or custom flow, I think it's clearer with something like `completePayment` like I used for Link. Or if you find a better naming please update on Link too so we keep it consistent.

Reply:
https://github.com/stripe/stripe-android/blob/master/link/src/main/java/com/stripe/android/link/LinkActivityContract.kt#L25-L35